proxyd/server.go

```diff
@@ -831,7 +831,15 @@ func (s *Server) genericRateLimitSender(ctx context.Context, tx *types.Transacti
 		return txpool.ErrInvalidSender
 	}
 
-	signer := types.LatestSignerForChainID(tx.ChainId())
+	var signer types.Signer
+	// If you pass in a zero chain ID, types.LatestSignerForChainID panics. So we need to handle that case
+	// manually.
+	if tx.ChainId().Cmp(common.Big0) == 0 {
+		signer = new(types.HomesteadSigner)
```

Comment:
So this wouldn't be an error case? How could a tx for a zero chain id be valid?

Reply:
if it's pre-EIP 155. the chain ID is zero if it's not replay-protected.